packages/generator/src/generator.js

```diff
@@ -359,7 +359,8 @@ export default class Generator {
       fileName = fileName === '/404/index.html' ? '/404.html' : fileName // /404 -> /404.html
     } else {
       const normalizedRoute = route.replace(/\/$/, '')
-      fileName = route.length > 1 ? path.join(path.sep, normalizedRoute + '.html') : path.join(path.sep, 'index.html')
+      const extension = (typeof this.options.generate.nonRootPageExtension === 'undefined') ? '.html' : this.options.generate.nonRootPageExtension
```

Comment:
`.html` can be directly put in `config/src/config/generate.js` as default option

Reply:
Yes, however for some reason some tests won't pass without that check

Reply:
I can help you on test when I'm available.

Reply:
BTW, you can easily solve your problem by using hook `generate:page`:

```js
// nuxt.config.js
export default {
  hooks: {
    generate: {
      page(page) {
        page.path = page.path.replace(/html^/, '...')
      }
    }
  }
}
```

Reply:
The original issue was not mine. Does this mean there is no need for such a configuration parameter?

Reply:
> The original issue was not mine

I'll leave comment there.

> Does this mean there is no need for such a configuration paramete

I would love to hear opinion from @Atinux @pi0

Reply:
I think so too. It is a really rare usage and it is possible to fix using hooks

Reply:
Using either hooks or postprocessing seems better option than adding a new feature.

Reply:
Sure, thanks.